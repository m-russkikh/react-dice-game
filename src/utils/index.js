import { DEFAULT_BALANCE } from '../constants/balance';

export function getPreloadedState() {
  let balance = {
    value: DEFAULT_BALANCE,
  };

  try {
    const serializedBalance = localStorage.getItem('balance');

    if (serializedBalance) {
      balance = JSON.parse(serializedBalance);
    }
  } catch (err) {
  //  ignore errors
  }

  return { balance };
}

export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getNextBetNumber() {
  return Math.floor(Math.random(0, 1) * 100) + 1;
}
