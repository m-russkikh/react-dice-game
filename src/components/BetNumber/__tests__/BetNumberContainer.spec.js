import { getShallowWrapper } from '../../../utils/testUtils';
import BetNumberContainer from '../containers/BetNumberContainer';


describe('BetNumberContainer', () => {
  let props;
  let wrapper;
  let instance;

  beforeEach(() => {
    props = {
      setBetNumber: jest.fn(),
    };

    wrapper = getShallowWrapper(BetNumberContainer, props);
    instance = wrapper.instance();
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    describe('handleChange', () => {
      it('should call setBetNumber', () => {
        const event = { currentTarget: { value: 5 } };

        instance.handleChange(event);

        expect(props.setBetNumber).toHaveBeenCalledTimes(1);
        expect(props.setBetNumber).toHaveBeenCalledWith(event.currentTarget.value);
      });
    });
  });
});
