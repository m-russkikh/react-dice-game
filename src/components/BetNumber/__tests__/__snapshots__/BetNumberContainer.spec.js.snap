// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetNumberContainer render should display correct 1`] = `"<InputControlView label=\\"Number\\" onChange={[Function]} />"`;
